eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Gov.hs

```diff
@@ -243,12 +254,49 @@ actionWellFormed ga = failureUnless isWellFormed $ MalformedProposal ga
       ParameterChange _ ppd -> ppuWellFormed ppd
       _ -> True
 
+checkProposalsHaveAValidPrevious ::
+  forall era.
+  PrevGovActionIds era ->
+  GovSnapshots era ->
+  GovProcedures era ->
+  Test (ConwayGovPredFailure era)
+checkProposalsHaveAValidPrevious prevGovActionIds snapshots procedures =
+  let isValidMember ::
+        StrictMaybe (PrevGovActionId p (EraCrypto era)) ->
+        (GovAction era -> Bool) ->
+        (PrevGovActionIds era -> StrictMaybe (PrevGovActionId p (EraCrypto era))) ->
+        Bool
+      isValidMember prev snapshotCond prevActionAccessor =
+        all
+          (== Just True) -- lookup has to succeed _and_ purpose of lookedup action has to match condition
+          ( fmap (snapshotCond . gasAction)
+              . (flip Map.lookup . snapshotGovActionStates $ snapshots ^. curGovSnapshotsL @era)
+              . unPrevGovActionId
+              <$> prev
+          )
+          || prev == prevActionAccessor prevGovActionIds
+      isValid proposal = case proposal ^. pProcGovActionL of
```

Comment:
You can reuse this function here: 
https://github.com/input-output-hk/cardano-ledger/blob/b8c6c136657dc7282dc6fc899b790c28f6393a58/eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs#L340-L351
```suggestion
      isValid proposal = 
        prevActionAsExpected (proposal ^. pProcGovActionL) prevGovActionIds ||
        case proposal ^. pProcGovActionL of
```